### DISCUSSION TITLE
Change I2C Baud?

### DISCUSSION THREAD

**Discussion by ajw685** [MichaIng/DietPi#6457] [Q&A]
Hi there,

I'm trying to figure out how to update the baud rate for an i2c connection. Trying to move from default of 150000000 to 400000000 for accelerometer for Klipper (3d printing)
Rock64; DietPi; 
MPU-6050 on a GY-521 board.

```
sudo i2cdetect -y 0
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- 68 -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
```
68 is the MPU-6050 defined address so the Rock64 is at least seeing the accelerometer

For whatever reason, "clock-frequency = <100000>;" isn't included in the device tree overlay and this is the parameter that defines clock speed. (i think?)

I've been trying to decompile and repcompile dtb and am not having success.

Keep getting this even though I have decompiled dtbo into dts, edited to include clock frequency and recompiled.

```
dietpi@DietPi:~$ sudo /bin/enable_dtoverlay i2c0 i2c@ff150000 okay
Applying...
/dts-v1/;
/ {
        fragment@0 {
                target-path = "/i2c@ff150000";
                __overlay__ {
                        status="okay";
                };
        };
};
Applied!
```

**Comment by MichaIng** [comment]
The parameter seems to be correct: https://www.kernel.org/doc/html/latest/i2c/instantiating-devices.html

But the value is in Hz, hence for 400.000.000 Hz you'd need `<400000000>`? Do you otherwise see the overlay being applied in `/proc/device-tree`?
